java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/SearchController.java

```diff
@@ -125,13 +121,74 @@ public ResponseEntity<String> getWikiImages(@RequestParam String northWest, @Req
         return ResponseEntity.ok(gson.toJson(collection));
     }
     
+    @MultiPlatform
     @RequestMapping(path = {"/parse-image-info"}, produces = "application/json")
     @ResponseBody
-    public ResponseEntity<String> parseImageInfo(@RequestBody String data) {
-        Map<String, String> info = wikiService.parseImageInfo(data);
+    public ResponseEntity<String> parseImageInfo(@RequestBody(required = false) String data,
+                                                 @RequestParam(required = false) String imageTitle,
+                                                 @RequestParam(required = false) Long pageId,
+                                                 @RequestParam(required = false) String lang) throws IOException, SQLException {
+        if (imageTitle == null && data == null) {
+            return ResponseEntity.badRequest().body("Required imageTitle or data!");
+        }
+        
+        if (imageTitle == null) {
+            // old parsing
+            Map<String, String> info = wikiService.parseImageInfo(data);
+            return ResponseEntity.ok(gson.toJson(info));
+        }
+        
+        if (data != null) {
+            // save immediately to cache
+            wikiService.saveWikiRawDataToCache(imageTitle, pageId, data);
+        }
```

Comment:
The cache becomes vulnerable with this block.

Anyone now is able to poison the cache with any data.

For example, poison with URL(s) of fishing / porn sites and/or invalid content which could lead server to crash.

It is better to remove this block completely or add strong validation of the sender (token, deviceid, etc).